foundations/javascript_basics/fundamentals-5.md

```diff
@@ -28,47 +28,116 @@ As an example of what we mean, let's consider a `sumOfTripledEvens` function. It
+This is a much more elegant approach, what do you think? We could even go a step further and define a function right inside of `map()` like so:
 
-- `.filter` will take our `array` and remove all non-even numbers from it, so it the result will be `[4, 10, 2]`.
-- `.map` will take the result of `.filter` and then transform all those numbers into triple numbers, it's result will be `[12, 30, 6]`.
-- `.reduce` will take the result of `.map` and convert that array to a single value. The way it does this is by storing the sum in the `accumulator` variable, which we initialise to `0` as the second argument of the function. Then, it will iterate through the array returned by `.map` and add every element to `accumulator`, finally returning the accumulator.
-- Finally, it returns the output of `.reduce` from the function.
+```js
+const arr = [1, 2, 3, 4, 5];
+arr.map(num => num + 1); // Outputs [2, 3, 4, 5, 6]; 
+```
+
+Can you try think of a way to multiply every number in the array by 3 instead of adding 1 to it?  
+We could do it like this:
+```js
+arr.map(num => num * 3); // Outputs [6, 9, 12, 15, 18];
+```
 
-In the end, we have a function that is more concise, readable and less prone to bugs.
+#### The .filter() method
+
+`.filter(callback)` is very similar to `.map(callback)`, it still iterates through the array and applies the callback function on every item. However, now instead of simply transforming values, it evaluates them. 
+Let's say we had a function, `isOdd` that returns either `true` if a number is odd or `false` if it isn't. 
+
+The `.filter(callback)` expects the `callback` to return either `true` or `false`. If it returns `true`, the value is included in the output. Otherwise, it isn't.
+Consider the array from our previous example, `[1, 2, 3, 4, 5]`. 
+If we wanted to remove all even numbers from this array, we could use `.filter()` like this:
+
+```js
+arr.filter(isOdd); // Outputs [1, 3, 5];
+```
+
+- `.filter(isOdd)` will iterate through the `arr` and pass **every value** into the `isOdd()` callback. 
+- `.isOdd()` will return `true` when the value is odd, which means this value is included in the output.
+- If it's an even number, `isOdd()` will return `false` and not include it in the final output.
+
+#### The .reduce() method
+
+Finally, let's say that we wanted to multiply all of the numbers in our `arr` together like this: `1 * 2 * 3 * 4 * 5`.
+First, we'd have to declare a variable `total` and initialize it to 0. Then, we'd iterate through the array with a `for` loop and multiply the `total` by the current number.
+
+But we don't actually need to do all of that, we have our `.reduce(callback, initialValue)` method that will do the job for us. Just like `.map()` and `.reduce()` it expects a callback function.
+However, it also takes in an `initialValue` which makes our lives easier and we don't have to declare a variable like `const total = 1` and instead `.reduce()` does that for us.
```

Comment:
```suggestion
However, there are two key differences with this array method:
- The callback function takes two arguments instead of one. The first argument is the `accumulator`, which is the current value of the result *at that point in the loop*. The first time through, this. value will either be set to the `initialValue` (described in the next bullet), or the first element in the array if no `initialValue` is provided. The second argument for the callback is the `current` value, which is the item currently being iterated on.
- It also takes in an `initialValue` as a second argument (after the callback), which helps when we don't want our initial value to be the first element in the array. For instance, if we wanted to sum all numbers in an array, we could call reduce without an `initialValue`, but if we wanted to sum all numbers in an array and add 10, we could use 10 as our `initialValue`.
```